src/analysis/convex/strict.lean

```diff
@@ -0,0 +1,451 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import analysis.convex.basic
+import topology.algebra.mul_action
+import topology.algebra.ordered.basic
+
+/-!
+# Strictly convex sets
+
+This file defines strictly convex sets.
+
+A set is strictly convex if the open segment between any two distinct points lies in its interior.
+
+## TODO
+
+Define strictly convex spaces.
+
+Is there a better home for the pointwise topology lemmas?
+-/
+
+open set
+open_locale convex pointwise
+
+section has_continuous_mul
+variables {α : Type*} [topological_space α] [comm_group α] [has_continuous_mul α] {s t : set α}
+
+@[to_additive]
+lemma mul_interior_subset : s * interior t ⊆ interior (s * t) :=
+begin
+  rw subset_interior_iff_subset_of_open,
+  { rintro x ⟨a, b, ha, hb, rfl⟩,
+    exact set.mul_mem_mul ha (interior_subset hb) },
+  { rw ←set.Union_mul_left_image,
+    exact is_open_bUnion (λ x hx, (homeomorph.mul_left x).is_open_map _ is_open_interior) }
+end
+
+@[to_additive]
+lemma interior_mul_subset : interior s * t ⊆ interior (s * t) :=
+begin
+  rw subset_interior_iff_subset_of_open,
+  { rintro x ⟨a, b, ha, hb, rfl⟩,
+    exact set.mul_mem_mul (interior_subset ha) hb },
+  { rw ←set.Union_mul_right_image,
+    exact is_open_bUnion (λ x hx, (homeomorph.mul_right x).is_open_map _ is_open_interior) }
+end
+
+lemma interior_mul_interior_subset {s t : set α} :
+  interior s * interior t ⊆ interior (s * t) :=
+(set.mul_subset_mul_left interior_subset).trans interior_mul_subset
```

Comment:
See my comment in https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there.20code.20for.20X.3F/topic/Addition.20of.20interiors/near/263542378 ?

Reply:
Here is the golfed version : 
```lean
@[to_additive]
lemma mul_interior_subset : s * interior t ⊆ interior (s * t) :=
interior_maximal (set.mul_subset_mul_left interior_subset) is_open_interior.mul_left

@[to_additive]
lemma interior_mul_subset : interior s * t ⊆ interior (s * t) :=
interior_maximal (set.mul_subset_mul_right interior_subset) is_open_interior.mul_right
```
Note that for it to work you must fix `is_open.mul_[left, right]` by replacing the `topological_group` assumption by `has_continuous_mul` (the proof doesn't change).

Reply:
Ah damn, thank you!